.compute

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+
+# this script gets executed on every node of an allocation;
+# in our case we use the provided COMPUTE_* env variables
+# to construct a distributed TensorFlow cluster definition
+
+set -e
+
+# activating standard TensorFlow Python virtual environment
+source /usr/local/tensorflow/bin/activate
+
+# the standard script to execute
+base_cmd="bin/run-wer-automation.sh"
+
+# reading the comma separated node list into array "nodes"
+IFS=',' read -r -a nodes <<< "$COMPUTE_NODES"
+# keep fist node for convenience
+first=${nodes[0]}
+# hostname for debugging
+hostname=`hostname`
+
+if ((${#nodes[@]} == 1)); then
+    # there is only one (this) node - so we are alone and we don't need a cluster definition
+    echo "[slurm   ] Starting single node process on $hostname ..."
+    $base_cmd --job_name localhost "$@"
+else
+    # there is more than on node so we will build a cluster definition
+
+    # defining all cluster ports in a way that avoids collisions with other cluster allocations
+    # (that could eventually get scheduled on the same node)
+    ((port_base=10000 + (COMPUTE_JOB_NUMBER * 100) % 50000))
```

Comment:
It's not the numbers I'm questioning, just so much as if COMPUTE_JOB_NUMBER actually gets big enough so that that mod takes effect, won't it just definitely break if that's the number? It'd make more sense for the number to not be divisible by 100 so that when it wraps around, it might fill in the gaps that are likely to exist. I think an extra set of brackets around the mod would also help readability.

For example, if you had it wrap at 50050, it'd then use the next 50 ports that are unlikely to be used (and if you have more than 50 nodes per job, I guess tough luck for you). At least I think that's right anyway :) At the moment, you could take away the modulus and the effect will likely be the same (assuming that failure to bind ports kills a client).